spinalcordtoolbox/scripts/sct_maths.py

```diff
@@ -270,223 +315,465 @@ def __call__(self, _, namespace, values, option_string=None):
+    data = current_value.data.astype(float)
+    if current_value.data.ndim == 3 and dim == 't':
+        rms = np.abs(data)
+    else:
+        axis = ('x', 'y', 'z', 't').index(dim)
+        rms = np.sqrt(np.mean(np.square(data), axis))
+    return Image(rms, hdr=current_value.hdr)
+
+
+def apply_std(current_value, dim):
+    """Implementation for -std."""
+    return Image(
+        np.std(current_value.data, ('x', 'y', 'z', 't').index(dim), ddof=1),
+        hdr=current_value.hdr,
+    )
+
+
+def apply_bin(current_value, bin_thr):
+    """Implementation for -bin."""
+    return Image(
+        sct_math.binarize(current_value.data, bin_thr),
+        hdr=current_value.hdr,
+    )
+
+
+def apply_otsu(current_value, nbins):
+    """Implementation for -otsu."""
+    return Image(
+        sct_math.otsu(current_value.data, nbins),
+        hdr=current_value.hdr,
+    )
+
+
+def apply_adap(current_value, block_size, offset):
+    """Implementation for -adap."""
+    return Image(
+        sct_math.adap(current_value.data, block_size, offset),
+        hdr=current_value.hdr,
+    )
 
-        printv('\nDone! File created: ' + fname_out, verbose, 'info')
 
-    # Case 2: One or more array operations (array_in -> array_out)
+def apply_otsu_median(current_value, size, n_iter):
+    """Implementation for -otsu-median."""
+    return Image(
+        sct_math.otsu_media(current_value.data, size, n_iter),
+        hdr=current_value.hdr,
+    )
+
+
+def apply_percent(current_value, percentile):
+    """Implementation for -percent."""
+    return Image(
+        sct_math.perc(current_value.data, percentile),
+        hdr=current_value.hdr,
+    )
+
+
+def apply_thr(current_value, threshold):
+    """Implementation for -thr."""
+    return Image(
+        sct_math.threshold(current_value.data, lthr=threshold),
+        hdr=current_value.hdr,
+    )
+
+
+def apply_uthr(current_value, threshold):
+    """Implementation for -uthr."""
+    return Image(
+        sct_math.threshold(current_value.data, uthr=threshold),
+        hdr=current_value.hdr,
+    )
+
+
+def apply_dilate(current_value, size, *, shape, dim):
+    """Implementation for -dilate."""
+    return Image(
+        sct_math.dilate(current_value.data, size=size, shape=shape, dim=dim),
+        hdr=current_value.hdr,
+    )
+
+
+def apply_erode(current_value, size, *, shape, dim):
+    """Implementation for -erode."""
+    return Image(
+        sct_math.erode(current_value.data, size=size, shape=shape, dim=dim),
+        hdr=current_value.hdr,
+    )
+
+
+def apply_smooth(current_value, *sigmas):
+    """Implementation for -smooth."""
+    ndims = current_value.data.ndims
+    if len(sigmas) == 1:
+        sigmas *= ndims
+    elif len(sigmas) != ndims:
+        raise SctMathsValueError(f"expected 1 or {ndims} values, got {len(sigmas)} values")
+
+    # adjust sigma based on voxel size
+    pdims = current_value.dim[4:8]
+    sigmas = [s/d for s, d in zip(sigmas, pdims)]
```

Comment:
We should discuss at the SCT meeting whether this operation makes sense for 4D images.

Reply:
Again, I wish I had taken better notes at the SCT meeting, but from what I recall:
- These operations are indeed only meant for 3D images.
- We may eventually want them to be applied volume-wise to 4D images, but that will be another PR.